examples/audio/sound_effects.rs

```diff
@@ -0,0 +1,207 @@
+//! Sound effects are short audio clips which are played in response to an event: a button click, a character taking damage, footsteps, etc.
+//!
+//! In this example, we'll showcase a simple abstraction that can be used to load and play randomized sound effects in response to an event.
+//! The logic here is highly customizable: we encourage you to adapt it to meet your game's needs!
+
+use bevy::{
+    color::palettes::tailwind::{BLUE_600, BLUE_700, BLUE_800},
+    ecs::world::Command,
+    prelude::*,
+    utils::HashMap,
+};
+use rand::{distributions::Uniform, Rng};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        // This must be below the `DefaultPlugins` plugin, as it depends on the `AssetPlugin`.
+        .init_resource::<SoundEffects>()
+        .add_systems(Startup, spawn_button)
+        .add_systems(
+            Update,
+            (
+                play_sound_effect_when_button_pressed,
+                change_button_color_based_on_interaction,
+            ),
+        )
+        .run();
+}
+
+#[derive(Resource)]
+struct SoundEffects {
+    map: HashMap<String, Vec<Handle<AudioSource>>>,
```

Comment:
Would this not be a tiny bit simpler with a newtype? Imo the name `map` doesn't add anything here.